src/util/playlistApiCall.js

```diff
@@ -0,0 +1,89 @@
+import axios from "axios";
+
+// create a playlist
+export async function createPlaylistApi(singleVideo,title,playlistVideoDispatch,setToggle,toastDispatch) {
+  try {
+    const response = await axios({
+      method: "post",
+      url: "/api/user/playlists",
+      headers: { authorization: localStorage.getItem("token") },
+      data:{ playlist: { title: title } },
+    });
+         playlistVideoDispatch({
+         type: "CREATE_PLAYLIST",
+         payload:response.data.playlists,
+       });
+      const lengthOfPlaylist=(response.data.playlists.length)
+      const playlistID =response.data.playlists[lengthOfPlaylist-1]._id;
+      if(singleVideo){
+        addVideoToPlaylistApi(singleVideo,playlistID,playlistVideoDispatch)
+      }
+      //code to show toast here --->
+      toastDispatch({type: "PLAYLIST_CREATED"})
+      setToggle(true);
+    // saving the encodedToken in the localStorage
+  } catch (error) {
+    console.log("Error in creating a new playlist",error);
+  }
+
+}
+
+// add video in a playlist
+export async function addVideoToPlaylistApi(video,playlistId,playlistVideoDispatch,setToggle,toastDispatch) {
+  try {
+    await axios({
+      method: "post",
+      url: `/api/user/playlists/${playlistId}`,
+      headers: { authorization: localStorage.getItem("token") },
+      data:{ video },
+    });
+     //code to show toast here --->
+     toastDispatch({type: "VIDEO_ADDED"})
+     setToggle(true);
```

Comment:
this is not toggling ig, rather do setToggle(!toggle) or change the state name.

Reply:
i have fixed it ,thanks for the review